tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -310,6 +311,32 @@ class HloToLhloTensorStoreOpConverter
   }
 };
 
+/// The std.constant op generates tensors, which could be supplied to other
+/// xla_hlo ops. This conversion pattern replaces such std.constant operations
+/// by xla_lhlo.const ops.
+class StdToLhloConstantOpConverter : public BaseOpConversion<mlir::ConstantOp> {
+ public:
+  using BaseOpConversion<mlir::ConstantOp>::BaseOpConversion;
+  LogicalResult matchAndRewrite(
+      mlir::ConstantOp constOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter& rewriter) const final {
+    Operation* op = constOp.getOperation();
+    OpResult result = op->getResult(0);
+    RankedTensorType resultType = result.getType().dyn_cast<RankedTensorType>();
+    if (!resultType) return failure();
+
+    // Tensors generated by constant ops are always statically shaped.
+    CHECK(resultType.hasStaticShape());
```

Comment:
Seems like an assert() is more suitable in MLIR code.

Reply:
Done.